test/src/sdk-harness/test_projects/intrinsics/src/main.sw

```diff
@@ -0,0 +1,53 @@
+script;
+
+use std::intrinsics::*;
+use std::assert::assert;
+use std::constants::ZERO;
+
+struct TestStruct {
+    field_1: bool,
+    field_2: u64,
+}
+fn is_ref_type<T>(param: T) -> bool {
+    is_reference_type::<T>()
+}
+
+fn get_size_of<T>(param: T) -> u64 {
+    size_of::<T>()
+}
```

Comment:
I wonder if the actual library functions in the `intrinsics` module should do this?  They seem a bit redundant otherwise.  What is the motivation around creating the wrappers in the new module -- to hide the `__` prefix?

Reply:
> to hide the `__` prefix?

Yes, because the alternative is to make the specific names of the intrinsics part of the public API of the language. Which we could certainly do, but then it's one more set of keywords people have to account for.

>  But I don't anticipate needing to put anything else in this module particularly..?

Can't really think of anything either, but that's fine! Not every intrinsic needs to be available to the end user, as you said.

Reply:
I'm in favor of these wrappers to hide the `__` prefix. It helps keep a nice uniform api surface for users.